src/components/view/DocumentDetailView/DocumentDetailView.tsx

```diff
@@ -0,0 +1,157 @@
+      notesResultIdentifier: props.notesResultIdentifier,
+      surveyResultId: props.surveyResultId
+    };
+
+    if (props.preview) {
+      if (props.preview !== "PreviewLoading") {
+        const previewData = getPreviewData(props.preview);
+        setDocumentDetail(previewData);
+        return;
+      }
+    }
+
+    Promise.all([
+      MyDataHelps.getDeviceInfo(),
+      queryAllSurveyFiles({ ...params })
+    ]).then(([deviceInfo, uploadedFiles]) => {
+      if (uploadedFiles.length > 0) {
+        const file = uploadedFiles[0];
+        MyDataHelps.queryFiles({ category: file.fileCategory }).then(function (files) {
+          if (files.files.length === 0) {
+            setDocumentDetail({ ...file } as DocumentDetail);
+            return;
+          }
+
+          MyDataHelps.getFileDownloadUrl(files.files[0].key).then(function (response) {
+            let documentDetail: DocumentDetail = {
+              ...file,
+              presignedDocUrl: "",
+              presignedImageUrl: "",
+              showDownload: !["iOS", "Android"].includes(deviceInfo?.platform) && !file.fileName.endsWith("pdf"),
+              fileKey: files.files[0].key
+            };
+
+            if (isImageFile(file.fileName)) {
+              documentDetail.presignedImageUrl = response.preSignedUrl;
+            } else {
+              documentDetail.presignedDocUrl = response.preSignedUrl;
+            }
+
+            setDocumentDetail(documentDetail);
+          });
+        });
+      }
+    });
+  }
+
+  function isImageFile(fileName: string) {
+    if (!fileName) return false;
+
+    const imageExtensions = ['.jpg', '.jpeg', '.png', '.gif', '.bmp', '.webp', '.tiff', '.svg', '.ico', '.avif'];
+    const lowerCaseFileName = fileName.toLowerCase();
+    return imageExtensions.some(ext => lowerCaseFileName.endsWith(ext));
+  }
+
+  async function downloadFile() {
+    const deviceInfo = await MyDataHelps.getDeviceInfo();
+
+    if (!deviceInfo || deviceInfo.platform == "Web") {
+      window.open(documentDetail?.presignedDocUrl ?? documentDetail?.presignedImageUrl, "_blank");
+    } else {
+      (window as any).webkit.messageHandlers.OpenFile.postMessage({ url: documentDetail?.presignedDocUrl ?? documentDetail?.presignedImageUrl });
+    }
+  }
+
+  const deleteFile = async () => {
+    if (!documentDetail?.fileKey || !documentDetail?.surveyResultId) {
+      return;
+    }
+
+    if (window.confirm(language("delete-file-confirm"))) {
+      deleteSurveyResultFiles(documentDetail.surveyResultId, documentDetail.fileKey).then(() => {
+        setDocumentDetail(undefined);
+        MyDataHelps.dismiss();
+      });
+    }
+  };
+
+
+  useInitializeView(() => {
+    initialize();
+  }, [], []);
+
+  return (
+    <Layout colorScheme={props.colorScheme} >
+      {(!documentDetail) && <LoadingIndicator />}
+      {documentDetail?.showDownload &&
+        <div className="mdhui-survey-answer-file-download-button">
```

Comment:
Seems like this is unfinished?  Should be a nav bar with the "share" button on the top right and the back button so you can get back from here etc